src/rdymms.c

```diff
@@ -0,0 +1,398 @@
+// This code supports the C and Fortran MMS drivers and is not used in
+// mainline RDycore (though it is built into the library).
+
+#include <muParserDLL.h>
+#include <petscdmceed.h>
+#include <petscdmplex.h>
+#include <petscsys.h>
+#include <private/rdycoreimpl.h>
+#include <private/rdydmimpl.h>
+#include <private/rdymathimpl.h>
+
+// gravitational acceleration [m/s/s]
+static const PetscReal GRAVITY = 9.806;
+
+// NOTE: our boundary conditions are expressed in terms of momenta and not flow
+// velocities, so we have to chain together a few things to evaluate x and y
+// momenta.
+
+static PetscErrorCode SetSWEAnalyticBoundaryCondition(RDy rdy) {
+  PetscFunctionBegin;
+
+  // We only need a single Dirichlet boundary condition, populated with
+  // manufactured solution data.
+  static RDyFlowCondition analytic_flow = {
+      .name = "analytic_bc",
+      .type = CONDITION_DIRICHLET,
+  };
+  analytic_flow.height     = rdy->config.mms.swe.solutions.h;
+  analytic_flow.x_momentum = rdy->config.mms.swe.solutions.u;  // NOTE: must multiply by h when enforcing!
+  analytic_flow.y_momentum = rdy->config.mms.swe.solutions.v;  // NOTE: must multiply by h when enforcing!
```

Comment:
I believe the comment `// NOTE: must multiply by h when enforcing!` is referring to the code, which is a bit later in this file:

```
      boundary_values[3 * e + 1] = h[e] * u[e];
      boundary_values[3 * e + 2] = h[e] * v[e];
```
If that is the case, why not simply use/define `analytic_flow.u` instead of `analytic_flow.x_momentum`?

Reply:
Excellent question. This is just how we originally defined boundary conditions (in terms of momentum instead of flow velocity). If we're willing to change how we specify boundary conditions to use flow velocities instead (which I think would be fine), we wouldn't have to do this tricky stuff. Let me know what you think about this.

In any case, I'd prefer to switch how we do boundary conditions in a separate PR.